### PR TITLE
Update auth documentation

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

File: docs/google-cloud-auth.rst [hunk 1]
```diff
@@ -68,14 +68,8 @@ Credential Discovery Precedence
...
-the library will check properties of your local environment
-in the following order:
-
-#. Application running in Google App Engine
-#. JSON or PKCS12/P12 keyfile pointed to by
-   ``GOOGLE_APPLICATION_CREDENTIALS`` environment variable
-#. Credentials provided by the Google Cloud SDK (via ``gcloud auth login``)
-#. Application running in Google Compute Engine
+the library will check for credentials in your environment by following the
+precedence outlined by :func:`google.auth.default`.
```
Comment (dhermes) [inline docs/google-cloud-auth.rst]: Intersphinx updated yet?

File: docs/google-cloud-auth.rst [hunk 2]
```diff
@@ -93,10 +87,9 @@ However, you may want to be explicit because
   from different projects
 
 In these situations, you can create an explicit
-:class:`Credentials <oauth2client.client.Credentials>` object suited to your
-environment.
-After creation,
-you can pass it directly to a :class:`Client <google.cloud.client.Client>`:
+:class:`Credentials <google.auth.credentials.Credentials>` object suited to your
```
Comment (dhermes) [inline docs/google-cloud-auth.rst]: `~google.auth.credentials.Credentials`?

File: docs/google-cloud-auth.rst [hunk 3]
```diff
@@ -192,10 +179,14 @@ After creation, :class:`Credentials <oauth2client.client.Credentials>`
 can be serialized with
 :meth:`to_json() <oauth2client.client.Credentials.to_json>`
 and stored in a file and then and deserialized with
-:meth:`from_json() <oauth2client.client.Credentials.from_json>`.
+:meth:`from_json() <oauth2client.client.Credentials.from_json>`. In order
+to use ``oauth2client``'s credentials with this library, you'll need to `convert
+them`_.
```
Comment (dhermes) [inline docs/google-cloud-auth.rst]: Put `convert them` on the same line pretty please

File: docs/google-cloud-auth.rst [hunk 4]
```diff
@@ -192,10 +179,14 @@ After creation, :class:`Credentials <oauth2client.client.Credentials>`
...
 
+.. _oauth2client: https://github.com/Google/oauth2client.
```
Comment (dhermes) [inline docs/google-cloud-auth.rst]: Capital-G-oogle?

File: docs/google-cloud-auth.rst [hunk 5]
```diff
@@ -337,10 +330,7 @@ using the `requests`_ library.
 .. _custom HTTP class: https://github.com/GoogleCloudPlatform/google-cloud-python/issues/908#issuecomment-110811556
 .. _requests: http://www.python-requests.org/en/latest/
 
-As for handling authentication on your own,
-it may be easiest just to re-use bits from ``oauth2client``.
-Unfortunately, these parts have a hard dependency on ``httplib2``.
-We hope to enable using `custom HTTP libraries`_ with ``oauth2client`` at
+We hope to enable using `custom HTTP libraries`_ with this library at
 some point.
```
Comment (dhermes) [inline docs/google-cloud-auth.rst]: some point sooon!

File: docs/google-cloud-auth.rst (hunk 1 above)
Comment (jonparrott) [inline docs/google-cloud-auth.rst]: Yeah it was updated in the previous PR.

Comment (dhermes) [inline docs/google-cloud-auth.rst]: Then I suck for not noticing / checking

File: docs/google-cloud-auth.rst (hunk 4 above)
Comment (jonparrott) [inline docs/google-cloud-auth.rst]: I just copy-pastad from github.

File: docs/google-cloud-auth.rst (hunk 2 above)
Comment (jonparrott) [inline docs/google-cloud-auth.rst]: Done.

**Pull request merged**